cmd/notary/integration_test.go

```diff
@@ -1526,3 +1527,447 @@ func TestWitness(t *testing.T) {
 		require.Error(t, err)
 	}
 }
+
+func TestClientTUFInitWithAutoPublish(t *testing.T) {
+	// -- setup --
+	setUp(t)
+
+	tempDir := tempDirWithConfig(t, "{}")
+	defer os.RemoveAll(tempDir)
+
+	server := setupServer()
+	defer server.Close()
+
+	tempFile, err := ioutil.TempFile("", "targetfile")
+	require.NoError(t, err)
+	tempFile.Close()
+	defer os.Remove(tempFile.Name())
+
+	var (
+		output       = ""
+		gun          = "MistsOfPandaria"
+		gunNoPublish = "Legion"
+
+		// This might be changed via the implementation, please be careful.
+		emptyList = "\nNo targets present in this repository.\n\n"
+	)
+	// -- tests --
+
+	// init repo with auto publish being enabled but with a malformed URL.
+	_, err = runCommand(t, tempDir, "-s", "For the Horde!", "init", "-p", gun)
+	require.Error(t, err, "Trust server url has to be in the form of http(s)://URL:PORT.")
+	// init repo with auto publish being enabled but with an unaccessible URL.
+	_, err = runCommand(t, tempDir, "-s", "https://notary-server-on-the-moon:12306", "init", "-p", gun)
+	require.NotNil(t, err)
+	require.Equal(t, err, nstorage.ErrOffline{})
+
+	// init repo with auto publish being enabled
+	_, err = runCommand(t, tempDir, "-s", server.URL, "init", "-p", gun)
+	require.NoError(t, err)
+	// list repo - expect empty list
+	output, err = runCommand(t, tempDir, "-s", server.URL, "list", gun)
+	require.NoError(t, err)
+	require.Equal(t, output, emptyList)
+
+	// init repo without auto publish being enabled
+	//
+	// Use this test to guarantee that we won't break the normal init process.
+	_, err = runCommand(t, tempDir, "-s", server.URL, "init", gunNoPublish)
+	require.NoError(t, err)
+	// list repo - expect error
+	_, err = runCommand(t, tempDir, "-s", server.URL, "list", gunNoPublish)
+	require.NotNil(t, err)
+	require.Equal(t, err, nstorage.ErrMetaNotFound{Resource: data.CanonicalRootRole})
+}
+
+func TestClientTUFAddWithAutoPublish(t *testing.T) {
+	// -- setup --
+	setUp(t)
+
+	tempDir := tempDirWithConfig(t, "{}")
+	defer os.RemoveAll(tempDir)
+
+	server := setupServer()
+	defer server.Close()
+
+	tempFile, err := ioutil.TempFile("", "targetfile")
+	require.NoError(t, err)
+	tempFile.Close()
+	defer os.Remove(tempFile.Name())
+
+	var (
+		output          = ""
+		target          = "ShangXi"
+		target2         = "ChenStormstout"
+		targetNoPublish = "Shen-zinSu"
+		gun             = "MistsOfPandaria"
+	)
+	// -- tests --
+
+	// init repo with auto publish being enabled
+	_, err = runCommand(t, tempDir, "-s", server.URL, "init", "-p", gun)
+	require.NoError(t, err)
+
+	// add a target with auto publish being enabled, but without the server URL
+	_, err = runCommand(t, tempDir, "add", "-p", gun, target, tempFile.Name())
+	require.NotNil(t, err)
+	require.Equal(t, err, nstorage.ErrOffline{})
+	// check status, since we only fail the auto publishment in the previous step,
+	// the change should still exists.
+	output, err = runCommand(t, tempDir, "status", gun)
+	require.NoError(t, err)
+
```

Comment:
Would it be worth checking that the output actually contains the first target here?  `notary status` returns no error if the changelists can be read, so an empty changelist would also not error.